tests/system/model_monitoring/test_app.py

```diff
@@ -1248,15 +1249,9 @@ def test_different_kind_of_serving(self) -> None:
             for endpoint in endpoints:
                 future = executor.submit(
                     self._test_endpoint,
-                    model_name=endpoint[mm_constants.EventFieldType.MODEL].split(":")[
-                        0
-                    ],
-                    feature_set_uri=endpoint[
-                        mm_constants.EventFieldType.FEATURE_SET_URI
-                    ],
-                    model_dict=self.router_models[
-                        endpoint[mm_constants.EventFieldType.MODEL].split(":")[0]
-                    ],
+                    model_name=endpoint.spec.model_name,
```

Comment:
here also you need to pass the endpoint name, you can also change the name of the function param for clarity. 
```suggestion
                    model_name=endpoint.metadata.name,
```